sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplate.java

```diff
@@ -1082,6 +1090,50 @@ private void applyBulkVersioning(Class<?> domainType,
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private <T, S extends T> PartitionKey getPartitionKeyFromValue(CosmosEntityInformation<T, ?> information, S entity) {
+        Object pkFieldValue = information.getPartitionKeyFieldValue(entity);
+        PartitionKeyBuilder partitionKeyBuilder = new PartitionKeyBuilder();
+        PartitionKey partitionKey;
+        if (pkFieldValue.getClass().isArray() || pkFieldValue instanceof Collection<?>) {
+            for (final Object pkValue : (ArrayList<Object>) pkFieldValue) {
```

Comment:
Same comment as above: 
Probably can reusing` PartitionKeyInternal fromObjectArray(Object[] values, boolean strict)` from the core SDK

Reply:
Same as above

Reply:
but since Spring SDK uses the azure-cosmos SDK underlying, so why can not use it?

Reply:
Done